pwiz_tools/Skyline/EditUI/SynchronizedIntegrationDlg.cs

```diff
@@ -38,6 +38,10 @@ public partial class SynchronizedIntegrationDlg : Form
         private readonly bool _originalAlignRtPrediction;
         private readonly ChromFileInfoId _originalAlignFile;
 
+        private int _idxLastSelected = -1;
+
+        private Tuple<string, HashSet<string>> _memory = Tuple.Create((string)null, new HashSet<string>());
```

Comment:
I would recommend making this a
Tuple&lt;ReplicateValue, IColllection&lt;object>>
You only need to convert things to strings if you need to persist them in Settings or something. If they only need to live for the life of dialog, you can keep everything as objects.

You can use "null" for the ReplicateValue for when they have not chosen a property.